File: 0009-palindrome-number/0009-palindrome-number.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
if(x<0||x%10===0&&x!=0){
    return false;
}
    
    let num=x;
    let reverse=0;
    while(num>0){
        let d=num%10;
       reverse = reverse * 10 + d; 
        num = Math.trunc(num / 10); 
    }
    return x===reverse;
};

File: 0009-palindrome-number/0009-palindrome-number.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x < 0 or (x % 10 == 0 and x != 0):
            return False
        
        num = x
        reverse = 0
        while num > 0:
            digit = num % 10
            reverse = reverse * 10 + digit
            num //= 10 
        return reverse == x
